lib/elixir/lib/file.ex

```diff
@@ -461,7 +461,7 @@ defmodule File do
   error will be returned.
   """
   @spec cp(Path.t, Path.t, (Path.t, Path.t -> boolean)) :: :ok | {:error, posix}
-  def cp(source, destination, callback \\ fn(_, _) -> true end) do
+  def cp(source, destination, callback \\ &path_differs?/2) do
```

Comment:
It feels that `path_differs?/2` check out of callback makes this function more rigid.
My thoughts are to let user decide how to handle existence of the destination file even if it's the source itself. I think we could have a public alternative to `path_differs?/2`. However, it's hard to say what the best for a user before we see it in use. :bowtie: